### PR TITLE
Add coldfront to ocp-staging

### PR DESCRIPTION
**Pull request by knikolla**
- [x] Include the coldfront base from github.com/nerc-project/coldfront-nerc
- [x] Use Postgres operator for database
- [x] Setup client and secret in Keycloak
- [x] ... in AWS
- [x] ... in external secrets
- [x] Figure out URLs and update in manifests

### CODE REVIEW

File: coldfront/overlays/ocp-staging/patches/deployments/coldfront-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: coldfront-deployment
+spec:
+  selector:
```
Comment (larsks) [inline coldfront/overlays/ocp-staging/patches/deployments/coldfront-deployment.yaml:6]: This selector element is provided by the base `kustomization.yaml` and does not need to be provided in the patch.

File: coldfront/overlays/ocp-staging/patches/deployments/coldfront-qcluster-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: coldfront-qcluster-deployment
+spec:
+  selector:
+    matchLabels:
+      app: coldfront
+      coldfront-qcluster-deployment: coldfront-app
```
Comment (larsks) [inline coldfront/overlays/ocp-staging/patches/deployments/coldfront-qcluster-deployment.yaml:9]: Is this label required? It doesn't appear to be used anywhere else, and if we don't need it we can remove this entire `selector` section. Without the patched selector, we get the following from the base repository:

```
   selector:
     matchLabels:
       app: coldfront
       coldfront-deployment: coldfront-qcluster-app
```

File: coldfront/overlays/ocp-staging/postgrescluster.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    - name: instance1
+      dataVolumeClaimSpec:
+        accessModes:
+          - "ReadWriteOnce"
+        resources:
+          requests:
+            storage: 1Gi
+  backups:
+    pgbackrest:
+      image: registry.developers.crunchydata.com/crunchydata/crunchy-pgbackrest:ubi8-2.38-2
```
Comment (larsks) [inline coldfront/overlays/ocp-staging/postgrescluster.yaml:19]: Check with @naved001 : do we need to provide an image here, or will the operator Do The Right Thing without this?

**Review by larsks — CHANGES_REQUESTED** [review]

File: coldfront/overlays/ocp-staging/postgrescluster.yaml (hunk 3 above)
Comment (naved001) [reply coldfront/overlays/ocp-staging/postgrescluster.yaml:19]: It should be left out and the operator will do the right thing, see: https://github.com/CCI-MOC/moc-apps/pull/178

Comment (knikolla) [reply coldfront/overlays/ocp-staging/postgrescluster.yaml:19]: Awesome, thanks, will update. The example included it so I left it there.

**Comment by knikolla** [comment]
Updated based on feedback.

**Review by larsks — APPROVED** [review]

**Review by naved001 — APPROVED** [review]

**Pull request merged**